### PR TITLE
(Port) MKUltra - Mind Control

### PR DESCRIPTION
**Pull request by xPokee**
## About The Pull Request
**Special thanks to waterpig and lessthanthree for helping me get this to work.**
Ports MKUltra from [Citadel](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/8356)

To summarize, this PR adds a new reagent, MKUltra.
MKU can be used to either make people your pet if they have the **_NEW ERP hypnosis interaction_** preference enabled or make people obey your commands if they don't have the pref enabled.
### How it's made

- one part coco powder
- one part bluespace dust
- one part psicodine
- one part mindbreaker toxin
- one part happiness
- one unit of blood (IT HAS TO BE FROM YOURSELF OR THE PERSON DRINKING IT WONT BE ENTHRALLED TO YOU)
### How it works (Taken directly from the preamble to the code at `modular_zubbers\code\modules\mkultra\MKUltra.dm`)
The magnum opus of FermiChem -
Long and complicated, I highly recommend you look at the two other files heavily involved in this
modular_zubbers\code\modules\mkultra\chem.dm - handles the subject's reactions
modular_zubbers\code\modules\mkultra\vocal_cords.dm - handles the enchanter speaking

HOW IT WORKS
chem.dm
There are 3 main ways this chemical works; I'll start off with discussing how it's set up.
Upon reacting with blood as a catalyst, the blood is used to define who the enthraller is - thus only the creator is/can choose who the master will be. As a side note, you can't adminbus this chem, even admins have to earn it.
This uses the fermichem only proc; FermiCreate, which is basically the same as On_new, except it doesn't require "data" which is something to do with blood and breaks everything, so I said bugger it and made my own proc. It basically sets up vars.
When it's first made, the creator has to drink some of it, in order to give them the vocal cords needed.
When it's given to someone, it gives them the status effect and kicks off that side of things. For every metabolism tick, it increases the enthrall tally.
Finally, if you manage to pump 100u into some poor soul, you overload them, and mindbreak them. Making them your willing, but broken slave. Which can only be reversed by; fixing their brain with mannitol and neurine (100 / 50u respectively) (or less with both),

vocal_cords.dm
This handles when the enchanter speaks - basically uses code from voice of god, but only for people with the status effect. Most of the words are self explanatory, and has a smaller range of commands. If you're not sure what one does, it likely affects the enthrall tally, or the resist tally.
list of commands:

-mixables-
enthrall_words
reward_words
punish_words
0
saymyname_words
wakeup_words
1
silence_words
antiresist_words
resist_words
forget_words
attract_words
2
sleep_words
strip_words
walk_words
run_words
knockdown_words
3
statecustom_words
custom_words
objective_words
heal_words
stun_words
hallucinate_words
hot_words
cold_words
getup_words
pacify_words
charge_words

Mixables can be used interspersed with other commands, 0 is commands that work on sleeper agents (i.e. players enthralled to state 3, then ordered to wake up and forget, they can be triggered back instantly)
1 is for players who immediately are injected with the chem - no stuns, only a silence and something that draws them towards them. This is the best time to try to fight it, and you're likely to win by spamming resist, unless the enchantress has plans.
2 is the seconds stage, which allows removal of clothes, slowdown and light stunning.
3 is the final stage, which allows application of a few status effects (see chem.dm) and allows custom triggers to be installed (kind of like nanites), again, see chem.dm
In a nutshell, this is the way you enthrall people, by typing messages into chat and managing cooldowns on the stronger words. You have to type words and your message strength is increases with the number of characters - if you type short messages the cooldown will be too much and the other player will overcome the chem.
I suppose people could spam gdjshogndjoadphgiuaodp but, the truth of this chem is that it mostly allows a casus belli for subs to give in, and everyone's a sub on cit (mostly), so if you aujigbnadjgipagdsjk then they might resist harder because you're a baddie and baddies don't deserve pets.
Also, the use of this chem as a murder aid is antithetic to its design, the subject gains bonus resistance if they're hurt or hungry (I'd like to expand this more, I like the idea that you have to look after all of them otherwise they aren't as effective, kind of like tamagachis!). If this becomes a problem, I'll deal with it, I'm not happy with people abusing this chem for an easy murder. (I might make it so you an't strike your pet when health is too low.)
Additionally, in lieu of previous statement - the pet is ordered to not kill themselves, even if ordered to.

chem.dm
oof
There are a few basic things that have to be understood with this status effect
1. There is a min loop which calculates the enthrall state of the subject, when the enthrall tally is over a certain amount, it will push you up 1 phase.
0 - Sleeper
1 - initial
2 - enthralled
3 - Fully entranced
4 - mind broken
4 can only be reached via OD, whereas you can increment up from 1 > 2 > 3. 0 is only obtainable on a state 3 pet, and it toggles between the two.

1.5 Chem warfare
Since this is a chem, it's expected that you will use all the chemicals at your disposal. Using aphro and aphro+ will weaken the resistance of the subject, while ananphro, anaphro+, mannitol and neurine will strengthen it.
Additionally, the more aroused you are, the weaker your resistance will be, as a result players immune to aphro and anaphro give a flat bonus to the enthralled.
The impure toxin for this chem increases resistance too, so if they're a bad chemist it'll be unlikely they have a good ratio (and as a secret bonus, really good chemists can purposely make the impure chem, to use either to combat the use of it against them, or as smoke grenades to deal with a large party)

2. There is a resistance proc which occurs whenever the player presses resist. You have to press it a lot, this is intentional. If you're trying to fight the enchanter, then you can't click both. You will usually win if you just mash resist and the enchanter does nothing, so you've got to react.
Each step further it becomes harder to resist, in state 2 it's longer, but resisting is still worthwhile. If you're not in state 3, and you've not got MKultra inside of you, you generate resistance very fast. So in some cases the better option will be to stall out any attempts to entrance you.
At the moment, resistance doesn't affect the commands - mostly because it's a way to tell if a state 3 is trying to resist. But this might change if it gets too hard to fight them off.
During state 3, it's impossible to resist if the enthraller is in your presence (8 tiles), you generate no resistance if so. If they're out of your range, then you start to go into the addiction processed
As your resistance is tied to your arousal, sometimes your best option is to wah

3. The addition process starts when the enthraller is out of range, it roughly follows the five stages of grief; denial, anger, bargaining, depression and acceptance.
What it mostly does makes you sad, hurts your brain, and sometimes you lash out in anger.
Denial - minor brain damaged
bargaining - 50:50 chance of brain damage and brain healing
anger - randomly lashing out and hitting people
depression - massive mood loss, stuttering, jittering, hallucinations and brain damage
depression, again - random stunning and crying, brain damage, and resistance
acceptance - minor brain damage and resistance.
You can also resist while out of range, but you can only break free of a stage 3 enthrallment by hitting the acceptance phase with a high enough resistance.
Finally, being near your enthraller reverts the damages caused.
It is expected that if you intend to break free you'll need to use psicodine and mannitol, or you'll end up in a bad, but not dead, state. This gives more work for medical!! Finally, the true rational of this complicated chem comes out.

4. Status effects in status effects.
There are a few commands that give status effects, such as antiresist, which will cause resistance presses to increase the enthrallment instead, theses are called from the vocal cords.
They're mostly self-explanatory; antiresist, charge, pacify and heal. Heals quite weak for obvious reasons. I'd like to add more, maybe some weak anneals with brute/exhaustion costs after the status is over. A truth serum might be neat too.
State 4 pets don't get status effects.

5. Custom triggers
Because it wasn't complicated enough already.
Custom triggers are set by stating a trigger word, which will call a sub proc, which is also defined when the trigger is Called
The effects available at the moment are:
Speak - forces pet to say a pre-allocated phrase in response to the trigger
Echo - sends a message to that player only (i.e. makes them think something)
Shock - gives them a seizure/zaps them
kneel - gives a short knockdown
strip - strips jumpsuit only
objective - gives the pet a new objective. This requires a high amount of mental capacity - which is determined by how much you resist. If you resist enough during phase 1 and 2, then they can't give you an objective.
Feel free to add more.
triggers work when said by ANYONE, not just the enchanter.
This is only state 3 pets, state 4 pets cannot get custom triggers, you broke them you bully.

7. If you're an antag you get a bonus to resistance AND to enthralling. Thus it can be worth using this on both sides. It shouldn't be hard to resist as an antag. There are further bonuses to command, Chaplains and chemist.
If you give your pet a collar then their resistance reduced too.
(I think that's everything?)

Fail states:
Creating the chemical with too low of purity or setting the temperature too high will cause an explosion equivalent to that of a meth explosion.

## Why It's Good For The Game
MKUltra gives both antagonists and erotic role players something to ask chemistry for, along with giving chemistry a new chemical to master making. Due to it requiring coco powder, it will additionally require the aid of botany, or going out of your way to get chocolate bars. But mostly, it enables more interactions between players, especially between antagonists and normal players, allowing for them to mind control players with more mechanics to back it up than just normal hypnosis does. _**I WOULD HIGHLY SUGGEST ANTAGONISTS USE LOOC BEFORE USING IT ON A VICTIM.**_

## Proof Of Testing
It has been testmerged on Towel for a few days now and everything seems to work as intended.

## Changelog

:cl: xPokee, LT3
add: added MKUltra, chemical enthrallment
/:cl:

### CODE REVIEW

**Comment by SapphoQueer** [comment]
bloodsucker consensual thralling: Now for non-bloodsuckers! But in a good way. I like this a lot actually. Very cool.

**Comment by StrangeWeirdKitten** [comment]
It's not a *critical* thing that will stop me from test merging it since I was told of the struggle you had trying to get this ported. 

In general practice, it's best to avoid single letter variables for code readibility like E.master, etc.

**Comment by xPokee** [comment]
> It's not a _critical_ thing that will stop me from test merging it since I was told of the struggle you had trying to get this ported.
> 
> In general practice, it's best to avoid single letter variables for code readibility like E.master, etc.

I'll run through them and give them all proper variables when I get the chance tomorrow, then.

**Comment by kassori** [comment]
The recipe for this is trivial, I can grow a plant with this in botany in 5 minutes.

**Comment by StrangeWeirdKitten** [comment]
> The recipe for this is trivial, I can grow a plant with this in botany in 10 minutes, please consider adding higher tier chems as reagents, see things like ghoul powder.

No, you don't have to do this.

**Comment by Swiftfeather** [comment]
As long as this has the same self heating and insane explosive power like it did on skyrat cit sounds good to me. Make chemistry dangerous again

**Comment by StrangeWeirdKitten** [comment]
> As long as this has the same self heating and insane explosive power like it did on skyrat cit sounds good to me. Make chemistry dangerous again

it does fam

File: modular_zubbers/code/modules/mkultra/MKUltra.dm [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+/datum/reagent/fermi/enthrall
+	name = "MKUltra"
+	description = "A forbidden deep red mixture that increases a person's succeptability to another's words. When taken by the creator, it will enhance the draw of their voice to those affected by it."
+	color = "#660015" // rgb: , 0, 255
+	taste_description = "synthetic chocolate, a base tone of alcohol, and high notes of roses"
+	overdose_threshold = 100 //If this is too easy to get 100u of this, then double it please.
+	metabolization_rate = 0.1//It has to be slow, so there's time for the effect.
+	data = list("creatorID" = null, "creatorGender" = null, "creatorName" = null)
+	var/creatorID  //ckey
```
Comment (Cyprex) [inline modular_zubbers/code/modules/mkultra/MKUltra.dm:137]: Just name this creatorKey imo

File: modular_zubbers/code/modules/mkultra/MKUltra.dm [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+If you give your pet a collar then their resistance reduced too.
+(I think thats everything?)
+
+Failstates:
+Blowing up the reaction produces a gas that causes everyone to fall in love with one another.
+
+Creating a chem with a low purity will make you permanently fall in love with someone, and tasked with keeping them safe. If someone else drinks it, you fall for them.
+*/
+
+/datum/reagent/fermi/enthrall
```
Comment (Cyprex) [inline modular_zubbers/code/modules/mkultra/MKUltra.dm:129]: I would personally name this typepath [...]/fermi/mkultra, although idk if the /fermi/ part even does anything these days!

**Comment by Mitryll** [comment]
I do not think it's a good idea to let this be used for actual mechanics at all. Leave it as a ERP chem for the people into it but this is blatantly too strong to be a normal tool to be used.

**Comment by Swiftfeather** [comment]
> I do not think it's a good idea to let this be used for actual mechanics at all. Leave it as a ERP chem for the people into it but this is blatantly too strong to be a normal tool to be used.

So true

**Comment by sneksnek** [comment]
Alright, I yapped on the discord server about this chemical for a while and will compress my thoughts here. The concept of this chemical could work, but with this design, it holds some issues. I think there are too many contingencies that would lead to griefing for it to be considered acceptable as an antagonist tool outside ERP. Here's a digestible list of reasons:

1. Enabling potential prefbreaking. I think this one is self-evident.
2. Potency for long term effects. Primarily deep-rooted traumas. These have some odd solutions but they're arduous to perform. As well, these effects are arguably too easy to inflict with this chemical.
3. It's overdesigned and doesn't currently inform victims how to counter the effects in-game. This could lead to some players gaining advantage on their victims by abusing the fact it's a custom chemical and hard to obtain information for. The original PR itself alludes to this obtuse complexity in a humorous manner.
4. Payloads. In order to inject large volumes of this chemical expediently for maximum effects, you must use a high volume foam or smoke payload. This ties into the other issues since this increases the probability of affecting multiple people and prefbreaking.

Some of these problems are ameliorated by categorizing it as an ERP chemical and thus not affecting those with preferences against it, but the ability to grief and prefbreak is still probable.

**Comment by LSommers** [comment]
Shouldn't have the SPLURT CODE tag, it's from Cit.

**Comment by Swiftfeather** [comment]
> Shouldn't have the SPLURT CODE tag, it's from Cit.

https://docs.google.com/spreadsheets/d/1hT1ITclikapnMb1LbVR8VGR2O0e71J76KN7PzvB_NDg/edit?gid=1937139898#gid=1937139898

**Comment by xPokee** [comment]
> 1. Enabling potential prefbreaking. I think this one is self-evident.

I debated for a few minutes over this one and I think the best way I can solve it is by making a whole new hypnosis preference, and then adding this to it once that preference would be merged. Though I am open to other ideas.
> 2. Potency for long term effects. Primarily deep-rooted traumas. These have some odd solutions but they're arduous to perform. As well, these effects are arguably too easy to inflict with this chemical.

I have yet to see someone get a deep-rooted trauma from this, if you have, please yet me know how they received it and I will make it so you can only get normal traumas from that source.
> 3. It's overdesigned and doesn't currently inform victims how to counter the effects in-game. This could lead to some players gaining advantage on their victims by abusing the fact it's a custom chemical and hard to obtain information for. The original PR itself alludes to this obtuse complexity in a humorous manner.

Currently you get a message about resisting but I can clarify that message more. Additionally if you try to resist once fully enthralled it WILL tell you how to do so, though at a cost to yourself.
> 4. Payloads. In order to inject large volumes of this chemical expediently for maximum effects, you must use a high volume foam or smoke payload. This ties into the other issues since this increases the probability of affecting multiple people and prefbreaking.

I was already on the fence about having the mindbroken (OD effect) state ported but I will definitely take this into consideration and potentially remove the OD effects entirely or set the OD threshold lower... fifty maybe?

**Comment by Mitryll** [comment]
> about

It gives the brain traumas/paralysis and other bad effects even if you have the enthralling pref off.

**Comment by Mitryll** [comment]
(Reposting cus had a formatting oopsie)

The chem gives brain traumas, paralysis and other super bad adverse effects even with the pref off that is waaaaay overboard compared to other combat chems.

**Comment by StrangeWeirdKitten** [comment]
It doesn't cause traumas through that method.

That being said, a correction I'll add is that when the pref is off, it just makes it standard non-lewd texts. It's being TM'd right now, so we'll assess it. How it works is that when the pref is off, it's a non-lewd version.

**Comment by kassori** [comment]
I'm not convinced what so ever that this is good for the game. It's entirely too simple to make while having massive, profound impacts on your character, which is already a rule violation, while also being so powerful it affects the rest of the station, since victims with inevitably interact with others and be weird.

It also doesn't matter if you disable the lewd messages with the preference off, because I am allowed to have my sex preferences on and still object, because I require active consent from people for anything sexual. If someone used ANY sex chem on me without my consent it would be a prefbreak.

**Comment by StrangeWeirdKitten** [comment]
I've directed XPokee to make a seperate lewd hypnosis prefrence for people who don't enjoy lewd hypnosis. If the prefrence properly filters out any lewd text, then it's not by nature a sexual thing, since it's not any diffrent to what a hypnoflash or brainwashing surgery can do (except you can actively resist out of it)

I'll be marking any broad claims of pref-breaking off topic, because that's not really an issue nor is it proper feedback. There's a stark diffrence between someone brainwashing you to kill the Head of Security versus someone brainwashing you with something much lewder when you have all your ERP hypnosis stuff set to no, which doesn't happen because someone is **actively** crossing those boundaries.

**Comment by Krypandenej** [comment]
Is there any reason why custom triggers and echoes wouldn't work (but custom objectives would)? That, and small request, please let us put additional Os and As in the bad job good job commands respectively

**Comment by xPokee** [comment]
> Is there any reason why custom triggers and echoes wouldn't work (but custom objectives would)?

I think they require the hypnosis preference (the new one) to be enabled by both the creator and the victim.

**Pull request merged**